src/main/java/com/hazelcast/remotecontroller/HzDockerCluster.java

```diff
@@ -0,0 +1,306 @@
+                    .withEnv("JAVA_OPTS", "-Dhazelcast.config=/opt/hazelcast/config_ext/hazelcast.xml")
+                    .withEnv("HZ_PHONE_HOME_ENABLED", "false")
+                    .withCopyFileToContainer(mountableFile, "/opt/hazelcast/config_ext/hazelcast.xml")
+                    .withNetwork(network)
+                    // For iptables to work, (at least for Mac) we need to run the container in privileged mode
+                    .withPrivilegedMode(true)
+                    .withExposedPorts(5701);
+        } else {
+            container = new GenericContainer(DockerImageName.parse(dockerImageString))
+                    .withEnv("HZ_PHONE_HOME_ENABLED", "false")
+                    .withNetwork(network)
+                    // For iptables to work, (at least for Mac) we need to run the container in privileged mode
+                    .withPrivilegedMode(true)
+                    .withExposedPorts(5701);
+        }
+        container.start();
+        if (!container.isRunning()) {
+            throw new RuntimeException("Container could not be started");
+        }
+
+        Integer port = container.getMappedPort(5701);
+        String host = container.getHost();
+        String containerId = container.getContainerId();
+
+        this.addContainer(container.getContainerId(), container);
+        tryInstallingIptables(container);
+        return new DockerMember(containerId, host, port);
+    }
+
+    private static void tryInstallingIptables(GenericContainer container) {
+        LOG.info("Installing iptables in the container.");
+        boolean installed = tryInstallingIptablesWithApk(container);
+        if (!installed) {
+            boolean installedWithApt = tryInstallingIptablesWithAptGet(container);
+            if(!installedWithApt) {
+                throw new RuntimeException("Could not install iptables in the container.");
+            }
+        }
+    }
+
+    private static boolean tryInstallingIptablesWithApk(GenericContainer container) {
+        try {
+            execAsRootAndThrowOnError(container, "apk", "update");
+            execAsRootAndThrowOnError(container, "apk", "add", "iptables");
+            return true;
+        } catch (Throwable e) {
+            LOG.error(e);
+            return false;
+        }
+    }
+
+    private static boolean tryInstallingIptablesWithAptGet(GenericContainer container) {
+        try {
+            execAsRootAndThrowOnError(container, "apt-get", "update");
+            execAsRootAndThrowOnError(container, "apt-get", "install", "iptables");
+            return true;
+        } catch (Throwable e) {
+            return false;
+        }
+    }
+
+    public boolean stopAndRemoveContainerById(String containerId) {
+        GenericContainer container = this.containers.get(containerId);
+        if (container == null) {
+            LOG.warn("Container does not exist with id: " + containerId);
+            return false;
+        }
+
+        container.stop();
```

Comment:
can this throw?

Reply:
I checked the code of it and apidoc, it cannot.